src/search.js

```diff
@@ -248,13 +326,86 @@ class Search {
         return re;
     }
 
+    $isMultilineSearch(options) {
+        return options.re && /\\r\\n|\\r|\\n/.test(options.re.source) && options.regExp && !options.$isMultiLine;
```

Comment:
options.$isMultiLine should be true in case if search contains \n.
Reference: https://github.com/ajaxorg/ace/blob/5e1e524d7ef04e13291b5a979fb4166e973e61f0/ace-internal.d.ts#L667C32-L667C34